packages/text-bitmap/src/BitmapText.ts

```diff
@@ -301,53 +331,199 @@ export class BitmapText extends Container
         }
 
         const lenChars = chars.length;
-        const tint = this.tint;
+
+        const pagesMeshData: Record<number, PageMeshData> = {};
+
+        // new Map<BaseTexture, PageMeshData>();
+        const meshesToAdd: Mesh[] = [];
 
         for (let i = 0; i < lenChars; i++)
         {
-            let c = this._glyphs[i]; // get the next glyph sprite
+            const texture = chars[i].texture;
+            const baseTextureUid = texture.baseTexture.uid;
 
-            if (c)
+            if (!pagesMeshData[baseTextureUid])
             {
-                c.texture = chars[i].texture;
+                let pageMeshData = pageMeshDataPool.pop();
+
+                if (!pageMeshData)
+                {
+                    const geometry = new MeshGeometry();
+                    const material = new MeshMaterial(Texture.EMPTY);
+
+                    const mesh = new Mesh(geometry, material);
+
+                    pageMeshData = {
+                        index: 0,
+                        indexCount: 0,
+                        vertexCount: 0,
+                        uvsCount: 0,
+                        total: 0,
+                        mesh,
+                        vertices: null,
+                        uvs: null,
+                        indices: null,
+                    };
+                }
+
+                // reset data..
+                pageMeshData.index = 0;
+                pageMeshData.indexCount = 0;
+                pageMeshData.vertexCount = 0;
+                pageMeshData.uvsCount = 0;
+                pageMeshData.total = 0;
+                // TODO need to get page texture here somehow..
+                pageMeshData.mesh.texture = new Texture(texture.baseTexture);
+                pageMeshData.mesh.tint = this._tint;
+
+                meshesToAdd.push(pageMeshData.mesh);
+
+                pagesMeshData[baseTextureUid] = pageMeshData;
             }
-            else
+
+            pagesMeshData[baseTextureUid].total++;
+        }
+
+        const currentMeshes = this._currentMeshes;
+
+        for (let i = 0; i < currentMeshes.length; i++)
+        {
+            if (meshesToAdd.indexOf(currentMeshes[i]) === -1)
             {
-                c = new Sprite(chars[i].texture);
-                c.roundPixels = this.roundPixels;
-                this._glyphs.push(c);
+                this.removeChild(currentMeshes[i]);
             }
+        }
+
+        for (let i = 0; i < meshesToAdd.length; i++)
+        {
+            if (meshesToAdd[i].parent !== this)
```

Comment:
Same here. `pagesMeshData` wasn't saved (initialized with `{}` inside `updateText`), hence all mesh-data (for each base-texture) was created again, so how can any mesh inside `meshesToAdd` have been a child of this BitmapText already?

Reply:
yeah, something smelly! going to maybe store pageMeshData instead of Mesh and we should be good..